go/enclave/rollupchain/rollup_chain.go

```diff
@@ -722,3 +737,29 @@ func (rc *RollupChain) verifySig(r *obscurocore.Rollup) bool {
 	pubKey := rc.storage.FetchAttestedKey(r.Header.Agg)
 	return ecdsa.Verify(pubKey, h[:], r.Header.R, r.Header.S)
 }
+
+// Retrieves the rollup with the given height, with special handling for earliest/latest/pending .
+func (rc *RollupChain) getRollup(height int) (*obscurocore.Rollup, error) {
+	var rollup *obscurocore.Rollup
+	switch gethrpc.BlockNumber(height) {
+	case gethrpc.EarliestBlockNumber:
+		rollup = rc.storage.FetchGenesisRollup()
+	case gethrpc.PendingBlockNumber:
+		// TODO - Depends on the current pending rollup; leaving it for a different iteration as it will need more thought.
+		return nil, fmt.Errorf("requested balance for pending block. This is not handled currently")
+	case gethrpc.LatestBlockNumber:
+		rollupHash := rc.storage.FetchHeadState().HeadRollup
+		maybeRollup, found := rc.storage.FetchRollup(rollupHash)
```

Comment:
Do you need the `maybeRollup` vars? I don't see any gotchas if you just use `rollup, found := ...` but could be wrong

Reply:
Fixed